nixos/lib/testing/nodes.nix

```diff
@@ -159,10 +160,11 @@ in
 
     passthru.nodes = config.nodesCompat;
 
-    defaults = mkIf config.node.pkgsReadOnly {
-      nixpkgs.pkgs = config.node.pkgs;
-      imports = [ ../../modules/misc/nixpkgs/read-only.nix ];
-    };
+    defaults = mkMerge [
+      (mkIf config.node.pkgsReadOnly {
+        imports = [ ../../modules/misc/nixpkgs/read-only.nix ];
+      })
+    ];
```

Comment:
Probably my suggestion was wrong. This undoes the optimization for tests that have `node.pkgsReadOnly = true;`; the opposite of your intent.
I think you can just revert this chunk, ie leave `nodes.nix` as is.

Reply:
Alright, reverted.